File: ARCADE/The Core/Level 4_1: leastFactorial.cs
    using System;
     
    public class Test
    {
    	public static int leastFactorial(int n){
            int gt = 1, i = 1;
            while (gt < n){
                gt = gt * i;
                i++;
            }
            return gt;
        }
    	public static void Main()
    	{
    		int n = 17;
    		Console.Write(leastFactorial(n));
    	}
    }

File: ARCADE/The Core/Level 4_1: leastFactorial.pas
program leastFactorial;

function leastFactorial(n: integer): integer;
var gt, i : integer;
begin
    gt := 1; i := 1;
    while(gt < n) do
    begin
        gt := gt * i;
        inc(i);
    end;
    leastFactorial := gt;
end;

var n : integer;
begin
	n := 17;
	writeln(leastFactorial(n));
end.
